scripts/docs-gen-snippets.mjs

```diff
@@ -0,0 +1,202 @@
+import fs from 'node:fs/promises';
```

Comment:
Could have been a ts file

Reply:
Was trying to follow the pattern that other scripts in this directory were written as js, but I'll go back and move to ts looping over any arrays with basically no type safety.

Reply:
I can sense a little frustration, if you think it has no value then fine 👍🏻 
Otherwise, you can use `Provider` type